src/app/pages/sharing/smb/smb-form/smb-form.component.ts

```diff
@@ -194,7 +199,12 @@ export class SmbFormComponent implements OnInit, AfterViewInit {
   form = this.formBuilder.group({
     path: ['', [Validators.required]],
     name: ['', Validators.required],
-    purpose: [null as SmbPresetType | null],
+    purpose: [SmbPresetType.DefaultShare as SmbPresetType | null, [
```

Comment:
After you close the dialog, the legacy type is still there on the list. If you don't touch the list, validation is not triggered and you can save share with legacy type.